notebooks/data-sources/oc-github-repo/github_PR_EDA.ipynb

```diff
@@ -2,7 +2,7 @@
  "cells": [
```

Comment:
<p>Looking at the data and the problem, my intuition is classification should be more useful here. From a development perspective, knowing whether a PR will take a couple of hours, half a day, full day, 2 days, a week would be a good assist. Trying to predict the exact number of hours with regression may be difficult to learn as well offer less utility. Binning may also to some extent help in reducing influence of external variables not included in the feature set (e.g. public holidays). Having said that, we'd have to define bins carefully and checkout ordinal classification methods. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/287/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Reply:
<p>@shreyanand I agree that intuitively, having a judgement of a time frame within which a PR will merge might offer more utility compared to having a knowledge of approximately when it merges to the hour(or another time unit). However, statistically, both seem like feasible approaches. To your point, ordinal classification could be useful here since the classes have a natural order and that's something we can explore in the following steps.</p>

Reply:
<p>@chauhankaranraj what are your thoughts on this?</p>

Reply:
<blockquote>regression may be difficult to learn</blockquote><p>Hmm at least based on the EDA here, I'm not sure if we have significant evidence to reject one of the two approaches (regression or classification).</p><p> </p><blockquote>offer less utility</blockquote><p>As far as utility goes, maybe yes, half-day vs full-day vs 2-days prediction is more meaningful than 4hr, 23hr, 45hr, but this seems like a subjective question. Since we don't know what the SMEs' preference is, it might be a bit premature rejecting classification or regression purely based on our notion of utility to SME.</p><p> </p><p>However, this got me thinking about end integration. And I feel like a classification setup might provide more flexibility in that. E.g. we can make something like XS/S/M/L/XL labels for time-to-merge, just as we have them for size of a PR. These can then be added to PRs by github bots. So this really makes me lean towards classification as the first exploration. Does that seem reasonable?</p>

Reply:
yes. let's move forward with a classification approach : ) and get this done. If in the future SME feedback suggests regression would be better, we can also come back and modify.